packages/react-router/lib/components.tsx

```diff
@@ -414,6 +417,79 @@ function DataRoutes({
   return useRoutes(routes || createRoutesFromChildren(children), location);
 }
 
+export interface DeferredResolveRenderFunction {
+  (args: { data: any }): JSX.Element;
+}
+
+export interface DeferredProps extends Omit<React.SuspenseProps, "children"> {
+  children: React.ReactNode | DeferredResolveRenderFunction;
+  data: any;
+  errorBoundary?: React.ReactNode;
```

Comment:
let's use `errorElement` so we don't have both `errorElement` on Route and `errorBoundary` on Deferred

Reply:
👍  [7504d99](https://github.com/remix-run/react-router/pull/9002/commits/7504d990bef9e4262187b4ceec6e577ab91b2e3b)